app/queries/users_with_access.rb

```diff
@@ -0,0 +1,23 @@
+class UsersWithAccess
```

Comment:
I don't like putting this in the model because you end up with massive god models that know about everything in the system.

Is `app/something` not a reasonably established convention now? `app/presenters` and `app/policies` are also non-standard, but used here. Pretty sure `app/queries` is used on other projects as well.